### PR TITLE
add tests for get requests for user & users

### PR DESCRIPTION
**Pull request by TolyK**

### CODE REVIEW

**Review by n-tolle — APPROVED** [review]
This looks good to me.  Sorry I don't have any suggested improvements this time.

File: test/test.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+chai.use(chaiHttp);
+
+describe('/GET users', () => {
+  it('it should GET all the users', (done) => {
+    chai.request(app)
+        .get('/users')
+        .end((err, res) => {
+              res.should.have.status(200);
+              res.body.data.should.be.a('array');
+              res.body.data.length.should.equal(10); //gt is greater than
```
Comment (LesliePajuelo) [inline test/test.js:18]: Brittle. What if you decide on an eleventh user?

just check that it's greater than 1

File: test/test.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+describe('/GET user', () => {
+  it('it should GET one user with ID 1', (done) => {
+    chai.request(app)
+        .get('/users/1')
+        .end((err, res) => {
+              res.should.have.status(200);
+              res.body.data.should.be.a('array');
+              res.body.data.length.should.equal(1);
+              res.body.data[0].user_name.length.should.be.gte(3);
+              res.body.data[0].user_name.should.be.a('string');
```
Comment (LesliePajuelo) [inline test/test.js:33]: Where are you mocking out this data?

File: test/test.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        });
+  });
+
+  it('it should not GET one user with ID 11', (done) => {
+    chai.request(app)
+        .get('/users/11')
+        .end((err, res) => {
+              res.should.have.status(400);
+              res.body.msg.should.be.a('string');
+              res.body.msg.should.equal('no user with id 11');
```
Comment (LesliePajuelo) [inline test/test.js:44]: Sufficient to test that it is type string with a length. Testing for this means doubling of work when you change the error message.

File: server/index.js [hunk 4]
```diff
@@ -33,6 +33,13 @@ app.get('/users/:id', async(req, res) => {
 
     const user = await db.getUser(id);
```
Comment (LesliePajuelo) [inline server/index.js:34]: What if there's an error?

**Comment by LesliePajuelo** [comment]
@n-tolle tagging you so you get some ideas for next time.

File: server/index.js [hunk 5]
```diff
@@ -33,6 +33,13 @@ app.get('/users/:id', async(req, res) => {
...
 
+    if ( user.length === 0 ) {
```
Comment (LesliePajuelo) [inline server/index.js:36]: without error handling on the db.getUser you'll get an error "cannot get length of undefined", which if it didn't stop execution at the very least is not truthy meaning it wouldn't go into this block and would pass to line 43

**Review by LesliePajuelo — COMMENTED** [review]
@n-tolle tagging you so you get some ideas for your next review.

@TolyK Good start to FEC.

**Pull request merged**